plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -784,6 +784,31 @@ public void testSnapshotReferenceSystemTable()
                         "('main', 'BRANCH', " + snapshotId3 + ", null, null, null)");
     }
 
+    @Test
+    public void testReadingSnapshotReference()
+    {
+        String tableName = "test_reading_snapshot_reference" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " WITH (partitioning = ARRAY['regionkey']) AS SELECT * FROM tpch.tiny.nation", 25);
+        Table icebergTable = loadTable(tableName);
+        long refSnapshotId = icebergTable.currentSnapshot().snapshotId();
+        icebergTable.manageSnapshots()
+                .createTag("test-tag", refSnapshotId)
+                .createBranch("test-branch", refSnapshotId)
+                .commit();
+
+        assertUpdate("INSERT INTO " + tableName + " SELECT * FROM tpch.tiny.nation LIMIT 5", 5);
+        assertQuery("SELECT * FROM " + tableName + " FOR VERSION AS OF " + refSnapshotId,
+                "SELECT * FROM nation");
+        assertQuery("SELECT * FROM " + tableName + " FOR VERSION AS OF 'test-tag'",
+                "SELECT * FROM nation");
+        assertQuery("SELECT * FROM " + tableName + " FOR VERSION AS OF 'test-branch'",
+                "SELECT * FROM nation");
+        assertQueryFails("SELECT * FROM " + tableName + " FOR VERSION AS OF 'test-wrong-ref'",
```

Comment:
nit: Add a new line before assertQueryFails to distinct positive and negative case.

Reply:
Done!